en/02_Developer_Guides/02_Controllers/07_CMS_JSON_APIs.md

```diff
@@ -0,0 +1,154 @@
+
+If you do include a message, remember that error messages are only intended for developers so do not use the `_t()` function to make them translatable. Do not use any returned messages on the frontend for things like toast notifications, instead those messages should be added directly in JavaScript.
+
+[info]
+Despite the slightly convoluted JSON format returned by `jsonError()` with multiple nodes, its usage remains consistent with `FormSchema`. It's better to use this method for uniformity rather than introducing separate methods for `FormSchema` and non-FormSchema failures.
+[/info.]
+
+## CSRF token
+
+When performing non-view operations, include an `'X-SecurityID'` header in your JavaScript request, with its value set to `SecurityToken::getSecurityID()`.
+
+Access the token value in JavaScript by logging in by using `import Config from 'lib/Config';` followed by `Config.get('SecurityID');`
+
+Ensure the security of your endpoints by validating the token header using `SecurityToken::inst()->checkRequest($this->getRequest())` on relevant endpoints. If this check fails then call `$this->jsonError(400);` which is consistent with the 400 status code used in [`FormRequestHandler::httpSubmission()`](api:SilverStripe\Forms\FormRequestHandler::httpSubmission()) when the CSRF check fails when submitting data using `FormSchema`.
+
+## Passing values from PHP to JavaScript
+
+To transmit values from PHP to JavaScript, override `LeftAndMain::getClientConfig()` within your controller. Begin your method with `$clientConfig = parent::getClientConfig();` to ensure proper inheritance.
+
+Include any relevant links to endpoints in the client configuration. For example, add `'myEndpointUrl' => $this->Link('my-endpoint')`, where `my-endpoint` is specified in `private static array $url_handlers`.
+
+In JavaScript, access these values using `import Config from 'lib/Config';` and retrieve the endpoint URL using `Config.getSection('Full\ClassName\Of\MyController').anArrayKey.myEndpointUrl;`.
+
+## JavaScript AJAX requests
+
+Use the `backend` helper which is a wrapper around `fetch()` when making JavaScript requests. Import it using `import backend from 'lib/Backend';`.
+
+The `backend` helper is able to use a `catch()` block to handle 400-500 response codes, offering a more streamlined approach compared to using vanilla `fetch()`. It also provides handy shorthand methods such as `.get()` and `.post()`, for writing concise code.
+
+The following code will make a POST request to an endpoint passing JSON data in the request body.
+
+```js
+// MyComponent.js
+
+const endpoint = `${Config.getSection(section).someKey.someUrl}`;
+const data = { somekey: 123 };
+const headers = { 'X-SecurityID': Config.get('SecurityID') };
+backend.post(endpoint, data, headers)
```

Comment:
Need import statement for `backend`, and a note that it's only available if using the webpack-config package

Reply:
Updated